activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -1,6 +1,6 @@
 require 'active_record/connection_adapters/abstract_mysql_adapter'
 
-gem 'mysql2', '>= 0.3.18', '< 0.5'
+gem 'mysql2', '>= 0.4.2', '< 0.5'
```

Comment:
Could leave the version dep as-is. Those who use Array flags will need 0.4.x or they'll get an error.

0.4.x is young still, worth keeping a more conservative version dep if possible.